app/models/claim/claims_preventing_payment_finder.rb

```diff
@@ -15,28 +15,21 @@ def initialize(claim)
     # The returned claims have different payment or tax details to those
     # provided by `claim`, and hence `claim` cannot be paid in the same payment
     # as the returned claims.
-    #
-    # NOTE: This only works for ECP/LUPP and TSLR cross policy as this requires a TRN
-    # Driven by: Policies.policies_claimable(policy) using OTHER_CLAIMABLE_POLICIES config otherwise this just returns []
     def claims_preventing_payment
       @claims_preventing_payment ||= find_claims_preventing_payment
     end
 
     private
 
     def find_claims_preventing_payment
-      return [] if claim.policy == Policies::InternationalRelocationPayments
-
-      eligibility_ids = claim.policy.policies_claimable.map { |policy|
-        policy::Eligibility.where(teacher_reference_number: claim.eligibility.teacher_reference_number)
-      }.flatten.map(&:id)
+      return [] if claim.personal_data_removed?
 
-      payrollable_claims_from_same_claimant = Claim.payrollable.where(eligibility_id: eligibility_ids)
+      payrollable_claims_from_same_claimant = Claim.payrollable.with_same_claimant(claim)
```

Comment:
I'm not sure if we need to preserve filtering the claims here by matching `policies_claimable` or not?